src/vmm/src/resources.rs

```diff
@@ -221,18 +230,41 @@ impl VmResources {
         self.block.insert(block_device_config)
     }
 
-    /// Inserts a network device to be attached when the VM starts.
-    pub fn set_net_device(
+    /// Builds a network device to be attached when the VM starts.
+    pub fn build_net_device(
         &mut self,
         body: NetworkInterfaceConfig,
     ) -> Result<NetworkInterfaceError> {
-        self.network_interface.insert(body)
+        self.net_builder.build(body).map(|net_device| {
+            // Update `Net` device `MmdsNetworkStack` IPv4 address.
+            match &self.mmds_config {
+                Some(cfg) => cfg.ipv4_addr().map_or((), |ipv4_addr| {
+                    if let Some(mmds_ns) = net_device.lock().unwrap().mmds_ns_mut() {
+                        mmds_ns.set_ipv4_addr(ipv4_addr);
+                    };
+                }),
+                None => (),
+            };
+        })
     }
 
     /// Sets a vsock device to be attached when the VM starts.
     pub fn set_vsock_device(&mut self, config: VsockDeviceConfig) -> Result<VsockConfigError> {
         self.vsock.insert(config)
     }
+
+    /// Settter for mmds config.
+    pub fn set_mmds_config(&mut self, config: MmdsConfig) {
+        // Update existing built network device `MmdsNetworkStack` IPv4 address.
+        for net_device in self.net_builder.iter_mut() {
+            config.ipv4_addr().map_or((), |ipv4_addr| {
+                if let Some(mmds_ns) = net_device.lock().unwrap().mmds_ns_mut() {
+                    mmds_ns.set_ipv4_addr(ipv4_addr);
+                };
+            });
+        }
```

Comment:
The API currently allows setting MMDS ipv4 to `None` aka use default, need to support setting it back to `None` as well.

```suggestion
        for net_device in self.net_builder.iter_mut() {
            if let Some(mmds_ns) = net_device.lock().unwrap().mmds_ns_mut() {
                mmds_ns.set_ipv4_addr(ipv4_addr);
            }
        }
```